librz/search/collection.c

```diff
@@ -5,10 +5,13 @@
 #include <rz_search.h>
 #include "search_internal.h"
 
-static RZ_OWN RzSearchCollection *rz_search_collection_new(RzSearchSpace space, RZ_NONNULL void *find, RZ_NONNULL RzSearchIsEmptyCallback is_empty, RZ_NULLABLE RzSearchFreeCallback free, RZ_NULLABLE void *user) {
+static RZ_OWN RzSearchCollection *rz_search_collection_new(RzSearchSpace space, RZ_NONNULL void *find, RZ_NONNULL RzSearchIsEmptyCallback is_empty, RZ_NULLABLE RzSearchFreeCallback free, RZ_OWN RZ_NULLABLE void *user) {
 	rz_return_val_if_fail(find && is_empty, NULL);
 	RzSearchCollection *sc = RZ_NEW0(RzSearchCollection);
 	if (!sc) {
+		if (free && user) {
+			free(user);
```

Comment:
Please rename the `free` parameter while you are at it. It shadows the free() from libc .